net/unbound/files/unbound.init

```diff
@@ -1,20 +1,718 @@
 #!/bin/sh /etc/rc.common
-# Copyright (C) 2016 Michael Hanselmann
+#
+##############################################################################
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+#   Copyright (C) 2016 Eric Luehrsen
+#
+##############################################################################
 
-START=61
+. /lib/functions/network.sh
 
+##############################################################################
+
+START=60
 USE_PROCD=1
+PROG="/usr/sbin/unbound"
+
+##############################################################################
+
+UNBOUND_B_CONTROL=0
+UNBOUND_B_DNSMASQ=0
+UNBOUND_B_DNSSEC=0
+UNBOUND_B_GATE_NAME=0
+UNBOUND_B_LOCL_BLCK=0
+UNBOUND_B_LOCL_NAME=0
+UNBOUND_B_LOCL_SERV=1
+UNBOUND_B_OBSCURE=0
+UNBOUND_B_PRIV_BLCK=1
+
+UNBOUND_D_RESOURCE=small
+UNBOUND_D_RECURSION=passive
+
+UNBOUND_TXT_FWD_ZONE=""
+UNBOUND_TTL_MIN=120
+
+UNBOUND_N_EDNS_SIZE=1280
+UNBOUND_N_FWD_PORTS=""
+UNBOUND_N_REC_PORT=53
+UNBOUND_N_ROOT_AGE=90
+
+UNBOUND_PID="/var/run/unbound.pid"
+UNBOUND_CONF="/var/etc/unbound.conf"
+UNBOUND_HEADER_CONF="/etc/unbound/header.conf"
+
+##############################################################################
+
+domain_ptr_ip6() {
```

Comment:
I'm commenting here not to point out a specific issue with this line, but because it's the first of a couple of rather complex functions. I get the feeling that you're trying too hard. There's too much magic going on. If you want to map unbound.conf settings to uci like dnsmasq and other daemons that's fine. Magic and voodoo not so much.

Feel free to split smaller improvements into individual pull requests. Make sure to not break backwards compatibility (within limits, of course). As it is I can't possibly accept this code, sorry.